ucsdscheduleme/Repo/CapeScrape.cs

```diff
@@ -0,0 +1,251 @@
+
+            // List of professor and courses that do not have cape review
+            List<Tuple<string, string>> profcourseNotFound = new List<Tuple<string, string>>();
+
+            // Iterate over all the courses we recieved from the database
+            foreach (var course in courses)
+            {
+                // for each course list out all the sections that are present
+                var sections = course.Sections.ToList();
+
+                // Variable to store all the professors that are teaching one course
+                List<string> listOfProfessors = new List<string>();
+
+                // Iterate over all the sections for each course
+                foreach (var section in sections)
+                {
+                    Professor currProfessor = section.Professor;
+                    // Update the Cape review for each professor only one time
+                    if (!listOfProfessors.Contains(currProfessor.Name))
+                    {
+                        listOfProfessors.Add(currProfessor.Name);
+
+                        // Note: Cape has already been initialized in the database model
+                        // as a empty list.
+
+                        // Gets the CAPE page for the specific professor
+                        string capePageURL = GenerateURL(currProfessor.Name, course.CourseAbbreviation);
+
+                        // If there is cape review page for the specific professor and specific course
+                        if (capePageURL.Length != 0)
+                        {
+                            ScrapeResult scrapedCapePage = InsertDataFromHtmlPage(capePageURL);
+
+                            if (scrapedCapePage != null)
+                            {
+                                // Updating the cape review result for professor object
+                                Cape newCapeReview = new Cape()
+                                {
+                                    Term = scrapedCapePage.Term,
+                                    StudentsEnrolled = scrapedCapePage.StudentsEnrolled,
+                                    NumberOfEvaluation = scrapedCapePage.NumberOfEvaluation,
+                                    RecommendedClass = scrapedCapePage.RecommendedClass,
+                                    RecommendedProfessor = scrapedCapePage.RecommendedProfessor,
+                                    StudyHoursPerWeek = scrapedCapePage.StudyHoursPerWeek,
+                                    AverageGradeExpected = scrapedCapePage.AverageGradeExpected,
+                                    AverageGradeReceived = scrapedCapePage.AverageGradeReceived,
+                                    URL = capePageURL
+                                };
+
+                                // Adding cape review to professor and cource object
+                                currProfessor.Cape.Add(newCapeReview);
+                                course.Cape.Add(newCapeReview);
```

Comment:
Don't you need to add the new cape to the cape table?

Reply:
I guess adding the cape to the course and prof and saving the context it creates the table.